prime-router/src/main/kotlin/azure/HistoryFunctions.kt

```diff
@@ -404,58 +403,61 @@ open class BaseHistoryFunction : Logging {
      * returns null if not authorized, otherwise returns a set of claims.
      */
     fun checkAuthenticated(request: HttpRequestMessage<String?>, context: ExecutionContext): AuthClaims? {
-        var userName = ""
-        // orgs in the settings table of the database have a format of "zz-phd",
-        // while the auth service claims has a format of "DHzz_phd"
-        // claimsOrgName will have the format of "DHzz_phd"
-        val claimsOrgName = request.headers["organization"] ?: ""
-
-        // orgName will have the format of "zz-phd" and is used to look up in the settings table of the database
-        var orgName = getOrgNameFromHeader(claimsOrgName)
-
-        var jwtToken = request.headers["authorization"] ?: ""
-
-        jwtToken = if (jwtToken.length > 7) jwtToken.substring(7) else ""
-
+        var userName: String? = null /* Format: email */
+        val requestOrgName = request.headers["organization"] ?: "" /* Format: xx-phd */
+        var oktaOrganizations: List<String?>
+        var accessOrgName: String? = null /* Format: xx-phd */
+        var jwtToken = request.headers["authorization"] ?: "" /* Format: Bearer ... */
+
+        /* INFO:
+        *   JWT cannot be parsed by OktaAuthentication object here because Admins who do not
+        *   exist in one or more DHxx_phd groups on Okta will be unable to see any receiver's
+        *   reports.
+        *
+        *   To fix this, below we apply the same verifier, and then we see if oktaOrganizations.contains(orgName) OR
+        *   oktaOrganizations.contains("DHPrimeAdmins") before authorizing.
+        */
         if (jwtToken.isNotBlank()) {
```

Comment:
Suggestion: Flip the logic? Early exiting a function if required data is missing is a best practice.

e.g.
```kotlin
if (!jwtToken.isNotBlank()) {
  return null;
}
```

Also, could you verify `userName`, `accessOrgName` and other required params here as well?

Basically "We expect all this data to be available and if it isn't just fail the function early.

Also, that both `isNullOrBlank()` and `isNotBlank()` are used seems odd.

Was there a motive to changing `userName` and `accessOrgName` defaults from `""` to `null`?

Reply:
I agree with the early exit refactor, I typically like that structure more, this was just already written in this order. As for the checks on `userName` and `accessOrgName`, they'll remain null until the JWT is parsed, so those checks can't be qualified at the beginning of the function. 

The use of `!x.isNotBlank()` is another leftover from the original code, but for context, those string methods can be weird. There is `isBlank()` `isNull()` `isNullAndBlank` as well as `isNotBlank` but no `isNotNull` or `isNotNullOrBlank` ... those seem to be relegated to `isNotEmpty`, BUT there is a corresponding `isEmpty` and it seems to be slightly different than `isBlank`, so often I have to get a little creative with nullable strings.